### PR TITLE
Add InstanceImageNav v3 dataset for Navigation Challenge

### PR DESCRIPTION
**Pull request by jacobkrantz**
## Motivation and Context

This PR adds InstanceImageNav dataset v3 to be used with the 2023 Habitat Navigation Challenge. This dataset version is similar to v2 with the primary difference that all episodes can successfully be navigated without traversing between floors. 

Related PR for the ObjectNav dataset: https://github.com/facebookresearch/habitat-lab/pull/1185

Note: `datasets/instance_imagenav/hm3d_v3.yaml` will be contributed separately.

## How Has This Been Tested

Tested locally by training an ImageNav agent.

## Types of changes

- **\[Docs change\]** Addition or changes to the documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

**Comment by jacobkrantz** [comment]
@ykarmesh

File: DATASETS.md [hunk 1]
```diff
@@ -27,8 +27,9 @@ These datasets can be downloaded follow the instructions [here](https://github.c
...
+| [Instance image goal navigation](https://arxiv.org/abs/2211.15876) | HM3DSem-v0.1 | [instance_imagenav_hm3d_v1.zip](https://dl.fbaipublicfiles.com/habitat/data/datasets/imagenav/hm3d/v1/instance_imagenav_hm3d_v1.zip) | `data/datasets/instance_imagenav/hm3d/v1/` |  [`datasets/instance_imagenav/hm3d_v1.yaml`](habitat-lab/habitat/config/habitat/dataset/instance_imagenav/hm3d_v1.yaml) | 303 MB |
+| [Instance image goal navigation](https://arxiv.org/abs/2211.15876) | HM3DSem-v0.2 | [instance_imagenav_hm3d_v2.zip](https://dl.fbaipublicfiles.com/habitat/data/datasets/imagenav/hm3d/v2/instance_imagenav_hm3d_v2.zip) | `data/datasets/instance_imagenav/hm3d/v2/` |  [`datasets/instance_imagenav/hm3d_v2.yaml`](habitat-lab/habitat/config/habitat/dataset/instance_imagenav/hm3d_v2.yaml) | 518 MB |
+| 🆕 [Instance image goal navigation](https://arxiv.org/abs/2211.15876) | HM3DSem-v0.2 | [instance_imagenav_hm3d_v3.zip](https://dl.fbaipublicfiles.com/habitat/data/datasets/imagenav/hm3d/v3/instance_imagenav_hm3d_v3.zip) | `data/datasets/instance_imagenav/hm3d/v3/` |  [`datasets/instance_imagenav/hm3d_v3.yaml`](habitat-lab/habitat/config/habitat/dataset/instance_imagenav/hm3d_v3.yaml) | 517 MB |
```
Comment (ykarmesh) [inline DATASETS.md:32]:
```suggestion
| 🆕 [Instance image goal navigation](https://arxiv.org/abs/2211.15876) | HM3DSem-v0.3 | [instance_imagenav_hm3d_v3.zip](https://dl.fbaipublicfiles.com/habitat/data/datasets/imagenav/hm3d/v3/instance_imagenav_hm3d_v3.zip) | `data/datasets/instance_imagenav/hm3d/v3/` |  [`datasets/instance_imagenav/hm3d_v3.yaml`](habitat-lab/habitat/config/habitat/dataset/instance_imagenav/hm3d_v3.yaml) | 517 MB |
```

Comment (jacobkrantz) [reply DATASETS.md:32]: That column is referencing the scene dataset, which is still `HM3DSem-v0.2` for the v3 IIN episode dataset.

Comment (ykarmesh) [reply DATASETS.md:32]: Oh okay got it

**Review by ykarmesh — APPROVED** [review]

**Pull request merged**